codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java

```diff
@@ -74,6 +77,9 @@ public HttpHeaders add(HttpHeaders headers) {
     @Override
     public HttpHeaders set(HttpHeaders headers) {
         if (headers instanceof DefaultHttpHeaders) {
+            if (headers == this) {
+                throw new IllegalArgumentException("can't add to itself.");
```

Comment:
hmm. when adding this check my thought process was that this should only be triggered by a coding error (i.e. couldn't think of a reason why one would need to do this).

either way is fine for me. I don't care deeply about this :-). Should be documented either way though.